### PR TITLE
Join nested with statements

### PR DESCRIPTION
**Pull request by vlopezferrando**
Join nested with statements into a single one. Rule [SIM117](https://github.com/MartinThoma/flake8-simplify/issues/35) from [flake8-simplify](https://github.com/MartinThoma/flake8-simplify).

Limitations:
* Only two nested with statements are joined.

### CODE REVIEW

File: .grit/patterns/join_nested_withs.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+# Join with statements
+
+```python
+with open("file1.txt") as f1:
+    with open("file2.txt", "r+") as f2:
+        pass
+
+# TODO: should be joined into a single with
```
Comment (morgante) [inline .grit/patterns/join_nested_withs.md:38]: Shouldn't we also handle this case?

```
with A() as a, B() as b:
  with C() as C:
    pass
```

Once we do, this case is easily handled.

**Review by morgante — CHANGES_REQUESTED** [review]

File: .grit/patterns/join_nested_withs.md (hunk 1 above)
Comment (vlopezferrando) [reply .grit/patterns/join_nested_withs.md:38]: With the latest changes,

```
with A() as a, B() as b:
    with C() as c:
        pass
```

Is correctly converted to:

```
with A() as a, B() as b, C() as c:
    pass
```

But I still can't join three `with`s one inside the other:

```
with A() as a:
    with B() as b:
        with C() as c:
            pass
```

Comment (morgante) [reply .grit/patterns/join_nested_withs.md:38]: Thanks, that's somewhat to be expected without recursive pattern support.

Comment (vlopezferrando) [reply .grit/patterns/join_nested_withs.md:38]: Ok!

**Pull request merged**